### PR TITLE
Isort as a pre-commit hook

### PR DESCRIPTION
**Pull request by ebadkamil**
Added `isort` as a `pre-commit` hook. Is it ok?

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -11,4 +11,10 @@ repos:
 -   repo: https://github.com/pre-commit/mirrors-mypy.git
     rev: v0.790
     hooks:
-    -   id: mypy
+    - id: mypy
+-   repo: https://github.com/pycqa/isort
+    rev: 5.7.0
+    hooks:
+    - id: isort
+      args: ["--profile", "black"]
```
Comment (ebadkamil) [inline .pre-commit-config.yaml:19]: to resolve conflict between black and isort 

https://github.com/PyCQA/isort/issues/1518#issuecomment-703450843

**Comment by mattclarke** [comment]
I'm pretty certain this is fine, so if @matthew-d-jones is happy I say we merge it despite the (known) issue with the build.

**Review by matthew-d-jones — APPROVED** [review]
Looks good, fine with me.
The problem is lack of disk space on system test node? Is someone dealing with that?

**Comment by matthew-d-jones** [comment]
@ebadkamil Sorry, after merging your other PR there are some merge conflicts to solve.

**Comment by ebadkamil** [comment]
Now worries. Now conflicts are fixed

**Review by matthew-d-jones — APPROVED** [review]

**Pull request merged**